src/tools/rbd/action/TrashPurgeSchedule.cc

```diff
@@ -0,0 +1,712 @@
+      std::cerr << "rbd: invalid schedule status JSON received" << std::endl;
+      return -EBADMSG;
+    }
+
+    try {
+      auto &s = json_root.get_obj();
+
+      if (s["scheduled"].type() != json_spirit::array_type) {
+        std::cerr << "rbd: unexpected schedule JSON received: "
+                  << "scheduled is not array" << std::endl;
+        return -EBADMSG;
+      }
+
+      for (auto &item_val : s["scheduled"].get_array()) {
+        if (item_val.type() != json_spirit::obj_type) {
+          std::cerr << "rbd: unexpected schedule status JSON received: "
+                    << "schedule item is not object" << std::endl;
+          return -EBADMSG;
+        }
+
+        auto &item = item_val.get_obj();
+
+        if (item["schedule_time"].type() != json_spirit::str_type) {
+          std::cerr << "rbd: unexpected schedule JSON received: "
+                    << "schedule_time is not string" << std::endl;
+          return -EBADMSG;
+        }
+        auto schedule_time = item["schedule_time"].get_str();
+
+        if (item["pool_name"].type() != json_spirit::str_type) {
+          std::cerr << "rbd: unexpected schedule JSON received: "
+                    << "pool_name is not string" << std::endl;
+          return -EBADMSG;
+        }
+        auto pool_name = item["pool_name"].get_str();
+
+        if (item["namespace"].type() != json_spirit::str_type) {
+          std::cerr << "rbd: unexpected schedule JSON received: "
+                    << "namespace is not string" << std::endl;
+          return -EBADMSG;
+        }
+        auto namespace_name = item["namespace"].get_str();
+
+        scheduled.push_back({schedule_time, pool_name, namespace_name});
+      }
+
+    } catch (std::runtime_error &) {
+      std::cerr << "rbd: invalid schedule JSON received" << std::endl;
+      return -EBADMSG;
+    }
+
+    return 0;
+  }
+
+  void dump(Formatter *f) {
+    f->open_array_section("scheduled");
+    for (auto &item : scheduled) {
+      f->open_object_section("item");
+      f->dump_string("schedule_time", item.schedule_time);
+      f->dump_string("pool", item.pool_name);
+      f->dump_string("namespce", item.namespace_name);
+      f->close_section(); // item
+    }
+    f->close_section(); // scheduled
+  }
+
+  friend std::ostream& operator<<(std::ostream& os, ScheduleStatus &d);
+
+private:
+
+  struct Item {
+    std::string schedule_time;
+    std::string pool_name;
+    std::string namespace_name;
+
+    Item(const std::string &schedule_time, const std::string &pool_name,
+         const std::string &namespace_name)
+      : schedule_time(schedule_time), pool_name(pool_name),
+        namespace_name(namespace_name) {
+    }
+  };
+
+  std::list<Item> scheduled;
+};
+
+std::ostream& operator<<(std::ostream& os, ScheduleStatus &s) {
+  TextTable tbl;
+  tbl.define_column("SCHEDULE TIME", TextTable::LEFT, TextTable::LEFT);
+  tbl.define_column("POOL", TextTable::LEFT, TextTable::LEFT);
```

Comment:
Nit: should we be listing pool and namespace as the first columns?

Reply:
The idea was that similarly to mirror snapshot scheduler, we wold have the table sorted by "schedule time". And in that case it would make sense to have this column first.

Now I am thinking that ordering by "schedule time", which I still think is useful in "mirror snapshot scheduler" case (to see what it going to be executed soon), probably is not very important here, as we are going to have much less records to display. So I think I am fine to change the column order.